go/worker/compute/txnscheduler/committee/node.go

```diff
@@ -139,9 +141,76 @@ func (n *Node) getMetricLabels() prometheus.Labels {
 
 // HandlePeerMessage implements NodeHooks.
 func (n *Node) HandlePeerMessage(ctx context.Context, message *p2p.Message) (bool, error) {
+	if message.TxnScheduleTx != nil {
+		n.logger.Debug("received peer message", "message", message)
+
+		expectedEpochNumber := message.TxnScheduleTx.ExpectedEpochNumber
+		call := message.TxnScheduleTx.Data
+
+		// Check if we are a leader. Note that we may be in the middle of a
+		// transition, but this shouldn't matter as the client will retry.
+		epochSnapshot := n.commonNode.Group.GetEpochSnapshot()
+		n.commonNode.CrossNode.Lock()
+		height := n.commonNode.CurrentHeight
+		n.commonNode.CrossNode.Unlock()
+		if !epochSnapshot.IsTransactionSchedulerLeader(height) {
+			n.logger.Error("unable to handle message, not leader anymore",
+				"err", api.ErrNotLeader,
+				"current_epoch", epochSnapshot.GetEpochNumber(),
+				"current_height", height,
+			)
+			return true, p2pError.Permanent(api.ErrNotLeader)
```

Comment:
This should probably return a `nil` error as you do want to relay the message instead of dropping it.